### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Resci**

### CODE REVIEW

File: src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoJdbcImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertQuery = "INSERT INTO manufacturers (name, country)"
```
Comment (AleksandrPok) [inline src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java:21]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc/checklist/05_JBDC_checklist.html
column naming

File: src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            preparedStatement.setString(1, manufacturer.getName());
+            preparedStatement.setString(2, manufacturer.getCountry());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getObject(1, Long.class));
+            }
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get connection in method create", e);
```
Comment (AleksandrPok) [inline src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java]:
```suggestion
            throw new DataProcessingException("Can't create manufacturer " + manufacturer, e);
```

File: src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        }
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String selectQuery = "SELECT * FROM manufacturers where id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement =
+                         connection.prepareStatement(
+                                 selectQuery, Statement.RETURN_GENERATED_KEYS)) {
```
Comment (AleksandrPok) [inline src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java]: Statement.RETURN_GENERATED_KEYS
it isn't necessary here

File: src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (!resultSet.next() || resultSet.getBoolean("deleted")) {
```
Comment (AleksandrPok) [inline src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java]: resultSet.getBoolean("deleted")) 
move this check into your query

File: src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+                return Optional.empty();
+            }
+            return Optional.of(parseFromResultSet(resultSet));
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get connection in method get", e);
```
Comment (AleksandrPok) [inline src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java]:
```suggestion
            throw new DataProcessingException("Can't get manufacturer by id: " + id, e);
```
rewrite other messages to make them more informative

File: src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    @Override
+    public List<Manufacturer> getAll() {
+        String selectQuery = "SELECT * FROM manufacturers";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement =
+                         connection.prepareStatement(selectQuery)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<Manufacturer> manufacturers = new ArrayList<>();
+            while (resultSet.next()) {
+                if (!resultSet.getBoolean("deleted")) {
```
Comment (AleksandrPok) [inline src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java]: same as a previous check

File: src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        }
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String selectQuery = "UPDATE manufacturers SET name = ?, country = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement =
+                         connection.prepareStatement(
+                                 selectQuery, Statement.RETURN_GENERATED_KEYS)) {
```
Comment (AleksandrPok) [inline src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java]: Statement.RETURN_GENERATED_KEYS
same

File: src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String selectQuery = "UPDATE manufacturers SET deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement preparedStatement =
+                         connection.prepareStatement(
+                                 selectQuery, Statement.RETURN_GENERATED_KEYS)) {
```
Comment (AleksandrPok) [inline src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java]: same

File: src/main/java/com/mate/exception/DataProcessingException.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
+package com.mate.exception;
+
+public class DataProcessingException extends RuntimeException {
+
+    public DataProcessingException(String message) {
```
Comment (AleksandrPok) [inline src/main/java/com/mate/exception/DataProcessingException.java:5]: Do you use this constructor anywhere?

**Review by AleksandrPok — APPROVED** [review]

File: src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+                 PreparedStatement preparedStatement =
+                         connection.prepareStatement(selectQuery)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<Manufacturer> manufacturers = new ArrayList<>();
+            while (resultSet.next()) {
+                manufacturers.add(parseFromResultSet(resultSet));
+            }
+            return manufacturers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get connection in method getAll", e);
```
Comment (RomanDubovskyi) [inline src/main/java/com/mate/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java]:
```suggestion
            throw new DataProcessingException("Can't get all manufacturers", e);
```

File: src/main/resources/taxi_schema.sql [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+ Source Host           : localhost:3306
+ Source Schema         : taxi_schema
+
+ Target Server Type    : MySQL
+ Target Server Version : 80022
+ File Encoding         : 65001
+
+ Date: 15/01/2021 01:09:47
+*/
+
```
Comment (RomanDubovskyi) [inline src/main/resources/taxi_schema.sql]: let's remove this comment

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**